debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbSourceInfoStructMaker.java

```diff
@@ -35,9 +35,14 @@ public Schema schema() {
     @Override
     public Struct struct(SourceInfo sourceInfo) {
         final Struct result = super.commonStruct(sourceInfo)
-                .put(SourceInfo.REPLICA_SET_NAME, sourceInfo.collectionId().replicaSetName())
-                .put(SourceInfo.COLLECTION, sourceInfo.collectionId().name())
-                .put(SourceInfo.ORDER, sourceInfo.position().getInc())
+                .put(SourceInfo.REPLICA_SET_NAME, sourceInfo.replicaSetName());
+
+        if (sourceInfo.collectionId() != null) {
+            result.put(SourceInfo.COLLECTION, sourceInfo.collectionId().name());
```

Comment:
@gunnarmorling I think schema needs to be updated as collection is mandatory.

Reply:
Is it really optional, though? IIUC, the recorded position in the oplog (offset) might denote an event that's not specific to a collection, but would the connector ever expose a CDC event which doesn't apply to a collection?

Reply:
@gunnarmorling The question is - either no-op does not come to this code - and then there is noo need for fix or it does and then it is optional.

Reply:
I think the root problem is one we had before: insufficient split of the logic for updating the source info based on the latest event received from the oplog and then "rendering" the source info struct. IIUC, the latter should only happen if we have a collection id at this point. I'll do some more digging.